src/modules/logicModule/BountyManager.sol

```diff
@@ -0,0 +1,528 @@
+        onlyClaimContributor(claimId)
+        validBountyId(bountyId)
+    {
+        validContributorsForBounty(contributors, _bountyRegistry[bountyId]);
+        Claim storage c = _claimRegistry[claimId];
+
+        uint length = c.contributors.length;
+        for (uint i; i < length; ++i) {
+            //remove ClaimId for each contributor address
+            contributorAddressToClaimIds[c.contributors[i].addr].remove(claimId); //@note c.contributors[i].addr -> is there a more gas efficient alternative to this?
+        }
+
+        delete c.contributors;
+
+        length = contributors.length;
+
+        for (uint i; i < length; ++i) {
+            c.contributors.push(contributors[i]);
+            //add ClaimId again to each contributor address
+            contributorAddressToClaimIds[contributors[i].addr].add(claimId);
+        }
+
+        emit ClaimContributorsUpdated(claimId, contributors);
+    }
+
+    /// @inheritdoc IBountyManager
+    function updateClaimDetails(uint claimId, bytes calldata details)
+        external
+        validClaimId(claimId)
+        onlyClaimContributor(claimId)
+    {
+        _claimRegistry[claimId].details = details;
+
+        emit ClaimDetailsUpdated(claimId, details);
+    }
+
+    /// @inheritdoc IBountyManager
+    function verifyClaim(uint claimId, uint bountyId)
+        external
+        onlyRole(uint8(Roles.VerifyAdmin))
+        validClaimId(claimId)
+        validBountyId(bountyId)
+        claimBelongingToBounty(claimId, bountyId)
+        notClaimed(bountyId)
+    {
+        Contributor[] memory contribs = _claimRegistry[claimId].contributors;
+
+        uint length = contribs.length;
+
+        //total amount needed to verifyBounty
+        uint totalAmount;
+
+        //current contributor in loop
+        Contributor memory contrib;
+
+        //For each Contributor add payments according to the claimAmount specified
+        for (uint i; i < length; i++) {
+            contrib = contribs[i];
+            totalAmount += contrib.claimAmount;
+
+            _addPaymentOrder(
+                PaymentOrder({
+                    recipient: contrib.addr,
+                    amount: contrib.claimAmount,
+                    createdAt: block.timestamp,
+                    dueTo: block.timestamp //dueTo Date is now
+                })
+            );
+        }
+
+        //ensure that this contract has enough tokens to fulfill all payments
+        _ensureTokenBalance(totalAmount);
+
+        //when done process the Payments correctly
+        __Module_proposal.paymentProcessor().processPayments(
+            IPaymentClient(address(this))
+        );
+
+        //Set completed to true
+        _bountyRegistry[bountyId].claimedBy = claimId;
+
+        emit ClaimVerified(claimId, bountyId);
+    }
+
+    //----------------------------------
+    // Role Functions
+
+    /// @inheritdoc IBountyManager
+    function grantBountyAdminRole(address addr) external onlyAuthorized {
```

Comment:
Should we put zero address checks here?

Reply:
Can you route a call via address(0) ?
If not there is no harm in adding the zero address to the role i think